wincode/src/io.rs

```diff
@@ -23,29 +17,79 @@ fn read_size_limit(len: usize) -> ReadError {
     ReadError::ReadSizeLimit(len)
 }
 
-impl<'a> Reader<'a> {
-    pub fn new(bytes: &'a [u8]) -> Self {
-        Self { cursor: bytes }
+/// Trait for structured reading of bytes from a source into potentially uninitialized memory.
+pub trait Reader<'a> {
+    /// A variant of the reader that can elide bounds checking.
+    ///
+    /// Useful for sections of code where there is enough information to bulk prefetch bytes
+    /// with a single bounds check for the entire chunk.
+    type Trusted: Reader<'a>;
+    /// Peek up to `n_bytes` bytes without consuming them.
+    ///
+    /// This is _not_ required to return exactly `n_bytes`, it is required
+    /// to return _up to_ `n_bytes`, so take care to check the length
+    /// if you need to ensure you get exactly `n_bytes`.
+    fn peek_buffered(&self, n_bytes: usize) -> &'a [u8];
```

Comment:
you don't like newlines?

not asking you to change - your crate your style, just telling you that I hate
it 😂